File: app/src/main/java/com/bigscreen/mangindo/content/MangaContentPagerAdapter.java
package com.bigscreen.mangindo.content;

import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class MangaContentPagerAdapter extends FragmentStatePagerAdapter {

    private List<MangaContentFragment> fragments;

    public MangaContentPagerAdapter(FragmentManager fm) {
        super(fm);
        fragments = new ArrayList<>();
    }

    @Override
    public MangaContentFragment getItem(int position) {
        if (getCount() > 0 && position < getCount()) {
            MangaContentFragment fragment = fragments.get(position);
            fragment.setFragmentPosition(position);
            return fragment;
        } else {
            return null;
        }
    }

    @Override
    public int getCount() {
        return fragments.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return "Page " + position;
    }

    public void setFragments(List<MangaContentFragment> fragments) {
        this.fragments.clear();
        this.fragments.addAll(fragments);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/bigscreen/mangindo/content/MangaContentPagerAdapter.kt
package com.bigscreen.mangindo.content

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class MangaContentPagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    var fragments: List<MangaContentFragment> = emptyList()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun getItem(position: Int): Fragment? {
        return if (position in 0 until count) {
            fragments[position].apply {
                setFragmentPosition(position)
            }
        } else {
            null
        }
    }

    override fun getCount(): Int = fragments.size

    override fun getPageTitle(position: Int): CharSequence = "Page $position"
}
